src/bin/overlay.rs

```diff
@@ -0,0 +1,123 @@
+//! # Overlay example
+//!
+//! This sample demonstrates how to create an element "floating" above others.
+
+extern crate gdk;
+extern crate gio;
+extern crate glib;
+extern crate gtk;
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use std::env::args;
+
+// Basic CSS: we change background color, we set font color to black and we set it as bold.
+const STYLE: &'static str = "
+#countdown-label {
+    background-color: rgba(192, 192, 192, 0.8);
+    color: black;
+    font-weight: bold;
+}";
+
+// upgrade weak reference or return
+#[macro_export]
+macro_rules! upgrade_weak {
+    ($x:ident, $r:expr) => {{
+        match $x.upgrade() {
+            Some(o) => o,
+            None => return $r,
+        }
+    }};
+    ($x:ident) => {
+        upgrade_weak!($x, ())
+    };
+}
+
+fn button_clicked(button: &gtk::Button, overlay_text_weak: &glib::object::WeakRef<gtk::Label>) {
+    let overlay_text = upgrade_weak!(overlay_text_weak);
+    overlay_text.set_text(&button.get_label().unwrap_or_else(|| String::new()));
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+
+    window.set_title("Overlay");
+    window.set_position(gtk::WindowPosition::Center);
+
+    window.connect_delete_event(|win, _| {
+        win.destroy();
+        Inhibit(false)
+    });
+
+    // The overlay container.
+    let overlay = gtk::Overlay::new();
+
+    // The overlay label.
+    let overlay_text = gtk::Label::new("0");
+    // We need to name it in order to apply CSS on it.
+    gtk::WidgetExt::set_name(&overlay_text, "countdown-label");
+    overlay_text.set_halign(gtk::Align::End);
+    overlay_text.set_valign(gtk::Align::Start);
+
```

Comment:
Why not centered?

Reply:
IMHO current `set_halign(gtk::Align::End);` is better that centered.

Reply:
Sounds better for the current case.